frontends/web/src/routes/settings/manage-accounts.tsx

```diff
@@ -273,36 +291,48 @@ class ManageAccounts extends Component<Props, State> {
                     </Button>
 
                     {
-                      accountsByKeystore.map(keystore => (<React.Fragment key={keystore.keystore.rootFingerprint}>
-                        <p>{keystore.keystore.name}</p>
-                        <div className="box slim divide m-bottom-large">
-                          { this.renderAccounts(keystore.accounts) }
-                        </div>
-                      </React.Fragment>))
+                      accountsByKeystore.map(keystore => {
+                        const currentAccount = keystore.accounts.find(account => account.code === editAccountCode);
+                        return (
+                          <div key={keystore.keystore.rootFingerprint}>
+                            <React.Fragment>
+                              <p>{keystore.keystore.name}</p>
+                              <div className="box slim divide m-bottom-large">
+                                {this.renderAccounts(keystore.accounts)}
+                              </div>
+                            </React.Fragment>
+                            <Dialog
+                              open={!!(editAccountCode)}
+                              onClose={() => this.setState({ editAccountCode: undefined, editAccountNewName: '', editErrorMessage: undefined })}
+                              title={t('manageAccounts.editAccountNameTitle')}>
+                              <form onSubmit={this.updateAccountName}>
+                                <Message type="error" hidden={!editErrorMessage}>
+                                  {editErrorMessage}
+                                </Message>
+                                <Input
+                                  onInput={e => this.setState({ editAccountNewName: e.target.value })}
+                                  value={editAccountNewName}
+                                />
+                                {
+                                  currentAccount && this.state.watchonly &&
+                                  <>
+                                    {this.renderWatchOnlyToggle(currentAccount)}
+                                  </>
+                                }
+                                <DialogButtons>
+                                  <Button
+                                    disabled={!editAccountNewName}
+                                    primary
+                                    type="submit">
+                                    {t('button.update')}
+                                  </Button>
+                                </DialogButtons>
+                              </form>
+                            </Dialog>
+                          </div>
+                        );
+                      })
```

Comment:
I tested with a 2nd wallet and got into a situation when it showed 2 dialogs 🤔  

![Screen Shot 2023-11-28 at 15 37 37](https://github.com/digitalbitbox/bitbox-wallet-app/assets/546900/c97875d3-c63f-4fe2-aeae-a1740373d6eb)

Reply:
Looks like the reason is that I have 2 items in `accountsByKeystore` and Dialog open condition is just if `editAccountCode` is truthy. This is the reason it opens a dialog for every keystore.